Userland/Libraries/LibGUI/AbstractView.cpp

```diff
@@ -110,8 +110,8 @@ void AbstractView::did_update_selection()
 {
     if (!model() || selection().first() != m_edit_index)
         stop_editing();
-    if (model() && on_selection && selection().first().is_valid())
-        on_selection(selection().first());
+    if (model() && on_selection_change)
+        on_selection_change();
```

Comment:
This will end up calling it twice, through
https://github.com/SerenityOS/serenity/blob/bd0a37fd26b4c0c78056a87e0a8de3185aafa074/Userland/Libraries/LibGUI/AbstractView.cpp#L197

Reply:
Thanks - fixed!